lstm/bmi_lstm.py

```diff
@@ -1,950 +1,562 @@
+INTERNAL_NAME_CROSSWALK = {
+    # dynamic inputs
+    "DLWRF_surface": "land_surface_radiation~incoming~longwave__energy_flux",
+    "PRES_surface": "land_surface_air__pressure",
+    "SPFH_2maboveground": "atmosphere_air_water~vapor__relative_saturation",
+    "APCP_surface": "atmosphere_water__liquid_equivalent_precipitation_rate",
+    "DSWRF_surface": "land_surface_radiation~incoming~shortwave__energy_flux",
+    "TMP_2maboveground": "land_surface_air__temperature",
+    "UGRD_10maboveground": "land_surface_wind__x_component_of_velocity",
+    "VGRD_10maboveground": "land_surface_wind__y_component_of_velocity",
+    # static inputs
+    "elev_mean": "basin__mean_of_elevation",
+    "slope_mean": "basin__mean_of_slope",
+    # outputs
+    "streamflow_cms": "land_surface_water__runoff_volume_flux",
+    "streamflow_m": "land_surface_water__runoff_depth",
+}
+"""
+Mapping from 'internal' names to 'external' names (names exposed via bmi).
+Internal names are meaningful to trained lstm models.
+Its healthy to think of internal names as aliases to external names.
+"""
+EXTERNAL_NAME_CROSSWALK = {v: k for k, v in INTERNAL_NAME_CROSSWALK.items()}
+"""Mapping from 'external' names to 'internal' names."""
+
+
+def crx_to_external(name: str):
+    """Return the external name (the name exposed via BMI) for a given internal name."""
+    return INTERNAL_NAME_CROSSWALK[name]
+
+
+def crx_to_interal(name: str):
+    """Return the internal name for a given external name (the name exposed via BMI)."""
+    return EXTERNAL_NAME_CROSSWALK[name]
+
+
+# ---------------  Ensemble Member -----------------------------
+
+
+class EnsembleMember:
+    """
+    An `EnsembleMember` is responsible for initializing and maintaining an LSTM model,
+    handling input scaling, managing hidden and cell states, and performing
+    inference using the trained model.
+    """
+
+    def __init__(self, cfg: dict[str, typing.Any], output_scaling_factor_cms: float):
+        self.cfg = cfg
+        # NOTE: aaraney: not sure if this *should* go here. leaving it for now.
+        self.output_scaling_factor_cms = output_scaling_factor_cms
+
+        # load training feature scales
+        scaler_file = cfg["run_dir"] / "train_data/train_data_scaler.yml"
+        with scaler_file.open("r") as fp:
+            train_data_scaler = yaml.safe_load(fp)
+        self.scalars = load_training_scalars(cfg, train_data_scaler)
+
+        # initialize torch lstm object
+        self.lstm = initialize_lstm(cfg)
+
+        # TODO: aaraney: how to handle input mapping conceptually?
+        # NOTE: this is order or variables in input tensor
+        self.input_names = cfg["dynamic_inputs"] + cfg["static_attributes"]
+
+        # WARNING: This implementation of the LSTM can only handle a batch size of 1
+        # No need to included different batch sizes
+        batch_size = 1
+        hidden_layer_size = cfg["hidden_size"]
+        # if init_config['initial_state'] == 'zero':
```

Comment:
@jmframe, can you have a look at this?

Reply:
In the [original code](https://github.com/NOAA-OWP/lstm/blob/9574d5b544ad4d321885fe29c0d2cc250eba346a/lstm/bmi_lstm.py#L311-L313) it looks like `h_t` and `c_t` are not initialized if `initial_state != 'zero'`. I could be missing something though.